storage/src/robocorp/storage/__init__.py

```diff
@@ -1,172 +1,275 @@
+    Raises:
+        AssetNotFound: Asset with the given name does not exist
+    """
+    LOGGER.info("Deleting asset: %s", name)
+    client = _get_client()
+    client.delete_asset(asset_id=f"name:{name}")
 
-    assets_client = _get_assets_client()
-    _handle_error = functools.partial(
-        _handle_missing_asset, name=name, asset_id=asset_id
-    )
-    response = assets_client.get(asset_id, _handle_error=_handle_error)
-    return cast(Asset, response.json())
 
+def _get_asset(name: str) -> "Response":
+    """Get an asset's payload URL."""
+    from ._requests import Requests
 
-def get_asset(name: str) -> str:
-    """Get an asset's value by providing its `name`.
+    client = _get_client()
+    details = client.get_asset(asset_id=f"name:{name}")
+    payload = details["payload"]
 
-    Args:
-        name: Name of the asset
+    if payload["type"] == "url":
+        url = payload["url"]
+        return Requests().get(url)
+    if payload["type"] == "empty":
+        raise ValueError(
+            f"Asset {details['name']!r} is empty."
+            + " It could mean an upload is still pending"
+            + " or has previously failed."
+        )
+    else:
+        # Note (2023-07-04):
+        # The 'data' payload type should only be used when uploading,
+        # and it should never be the the response for getting an asset
+        raise RuntimeError(f"Unsupported payload type: {payload['type']}")
+
+
+def get_text(name: str) -> str:
+    """Return the given asset as text.
+
+    Arguments:
+        name: Name of asset
 
     Returns:
-        The previously set value of this asset, or empty string if not set
+        Asset content as text
 
     Raises:
-        AssetNotFound: Asset with the given name does not exist
+        AssetNotFound: No asset defined with given name
     """
-    LOGGER.info("Retrieving asset: %r", name)
-    payload = _get_asset(name)["payload"]
-    if payload["type"] == "empty":
-        LOGGER.warning("Asset %r has no value set", name)
-        return ""
+    response = _get_asset(name)
+    return response.text
 
-    assets_client = _get_assets_client()
-    url = payload["url"]
-    return assets_client.get(url, headers={}).text
 
+def get_json(name: str, **kwargs) -> JSON:
+    """Return the given asset as a deserialized JSON object.
 
-def _create_asset(name: str) -> Asset:
-    LOGGER.debug("Creating new asset with name: %r", name)
-    assets_client = _get_assets_client()
-    body = {"name": name}
-    return assets_client.post("", json=body).json()
+    Arguments:
+        name: Name of asset
+        **kwargs: Additional parameters for `json.loads`
 
+    Returns:
+        Asset content as a Python object (dict, list, etc.)
 
-def set_asset(name: str, value: str, wait: bool = True):
-    """Creates or updates an asset named `name` with the provided `value`.
+    Raises:
+        AssetNotFound: No asset defined with given name
+        JSONDecodeError: Asset was not valid JSON
```

Comment:
I think because we use `.json()` from `requests`, it always guarantees a correct type comes from the response, but I'm not 100% sure because... well... Python exceptions don't have any typing like in Java or other languages.